kotlin/services/ec2/src/main/kotlin/com/kotlin/ec2/EC2Scenario.kt

```diff
@@ -0,0 +1,528 @@
+                    isRunning = true
+                }
+            }
+        }
+    }
+    return pubAddress
+}
+
+suspend fun runInstanceSc(instanceTypeVal: String, keyNameVal: String, groupNameVal: String, amiIdVal: String): String {
+    val runRequest = RunInstancesRequest {
+        instanceType = InstanceType.fromValue(instanceTypeVal)
+        keyName = keyNameVal
+        securityGroups = listOf(groupNameVal)
+        maxCount = 1
+        minCount = 1
+        imageId = amiIdVal
+    }
+
+    Ec2Client { region = "us-west-2" }.use { ec2 ->
+        val response = ec2.runInstances(runRequest)
+        val instanceId = response.instances?.get(0)?.instanceId
+        println("Successfully started EC2 Instance $instanceId based on AMI $amiIdVal")
+        return instanceId.toString()
+    }
+}
+
+// snippet-start:[ec2.kotlin.describe_instance_type.main]
+// Get a list of instance types.
+suspend fun getInstanceTypesSc(): String {
+    var instanceType = ""
+    val filterObs = ArrayList<Filter>()
+    val filter = Filter {
+        name = "processor-info.supported-architecture"
+        values = listOf("arm64")
+    }
+
+    filterObs.add(filter)
+    val typesRequest = DescribeInstanceTypesRequest {
+        filters = filterObs
+        maxResults = 10
+    }
+    Ec2Client { region = "us-west-2" }.use { ec2 ->
+        val response = ec2.describeInstanceTypes(typesRequest)
+        response.instanceTypes?.forEach { type ->
+            println("The memory information of this type is ${type.memoryInfo?.sizeInMib}")
+            println("Network information is ${type.networkInfo}")
+            instanceType = type.instanceType.toString()
+        }
+        return instanceType
+    }
+}
+// snippet-end:[ec2.kotlin.describe_instance_type.main]
+
+// Display the Description field that corresponds to the instance Id value.
+suspend fun describeImageSc(instanceId: String): String? {
+    val imagesRequest = DescribeImagesRequest {
+        imageIds = listOf(instanceId)
+    }
+
+    Ec2Client { region = "us-west-2" }.use { ec2 ->
+        val response = ec2.describeImages(imagesRequest)
+        println("The description of the first image is ${response.images?.get(0)?.description}")
+        println("The name of the first image is  ${response.images?.get(0)?.name}")
+
+        // Return the image id value.
+        return response.images?.get(0)?.imageId
+    }
+}
+
+// Get the Id value of an instance with amzn2 in the name.
+suspend fun getParaValuesSc(): String? {
+    val parameterRequest = GetParametersByPathRequest {
+        path = "/aws/service/ami-amazon-linux-latest"
+    }
+
+    SsmClient { region = "us-west-2" }.use { ssmClient ->
+        val response = ssmClient.getParametersByPath(parameterRequest)
+        response.parameters?.forEach { para ->
+            println("The name of the para is: ${para.name}")
+            println("The type of the para is: ${para.type}")

```

Comment:
These are a bit hard to read in the output. You may want to add a blank line in between each loop, or something to break them up and make them more readable.